src/DataSourceProvider.vue

```diff
@@ -1,19 +1,369 @@
+            <label for="showAll">
+              <span class="check">
+                <i class="fa fa-check"></i>
+              </span>
+              Show all data sources
+            </label>
+          </div>
+
+        </div>
+
+        <div v-else-if="selectedDataSource && !changeDataSource">
+          <p>{{ selectedDataSource.id }}. {{ selectedDataSource.name }} <a @click.prevent="onDataSourceChange" class=" change-data-source">Change</a></p>
+        </div>
+
+        <div v-show="selectedDataSource" @click="viewDataSource" class="btn btn-default btn-view-data-source">View data source</div>
+
+        <section v-show="selectedDataSource" class="security-notify">
+          <div v-if="!securityEnabled" class="alert alert-warning">
+            <p><b>This data source is missing security rules.</b></p>
+            <p>Configure security rules so the app can access the data</p>
+            <div @click="onAddDefaultSecurity" class="btn btn-primary btn-security">Configure security rules</div>
+          </div>
+          <div v-else-if="securityEnabled" class="alert alert-success">
+            Security rules added. To manage security rules click <b>View data source</b> above.
+          </div>
+        </section>
+
+      </section>
+    </div>
+  </section>
 </template>
 
 <script>
-import FirstComponent from './components/FirstComponent';
+import Select2 from './components/Select2.vue';
+import { getDataSources, getDataSource, createDataSource, updateDataSourceSecurityRules } from './services/dataSource';
 
 export default {
   data() {
-    return {};
+    return {
+      appDataSources: [],
+      allDataSources: [],
+      copyOfAllDataSources: [],
+      isLoading: true,
+      hasError: false,
+      errorMessage: '',
+      widgetData: {},
+      selectedDataSource: null,
+      dataSources: [],
+      changeDataSource: false,
+      securityEnabled: false,
+      showAll: false
+    };
+  },
+  methods: {
+    onDataSourceSelect(dataSource) {
+      this.selectedDataSource = dataSource;
+    },
+    initProvider() {
+      this.widgetData = Fliplet.Widget.getData();
+
+      if (this.widgetData.dataSourceId) {
+        return this.loadSelectedDataSource();
+      }
+
+      this.loadDataSources(this.widgetData.appId);
+    },
+    onAddDefaultSecurity() {
+      this.isLoading = true;
+      this.selectedDataSource.accessRules = this.widgetData.accessRules;
+
+      updateDataSourceSecurityRules(this.selectedDataSource.id, this.selectedDataSource.accessRules)
+        .then(() => {
+          Fliplet.Modal.alert({
+            message: 'Your changes have been applied to all affected apps.'
+          });
+        })
+        .catch(err => {
+          this.hasError = true;
+          this.errorMessage = Fliplet.parseError(err);
+        })
+        .finally(() => {
+          this.isLoading = false;
+        });
+    },
+    hasAccessRules: function() {
```

Comment:
Find all other named functions and correct them too.

```suggestion
    hasAccessRules() {
```